Warn when there are fewer teams than spaces in a schedule import

PeterJCLaw opened the pull request:
While this *is* supported, the logic around this somewhat guesswork and it turns out to be non-obvious to users how to override the default if that logic does not produce the desired result.

Example output:
``` console
$ srcomp import-schedule . SR2025-t29-a7.txt --extend 
Warning: more places than teams (30 places, 29 teams). Heuristics will be used to attempt to minimise the number of empty spaces in matches when mapping teams to places.
Consider using `--ignore-ids` to indicate which ids should be ignored if the default logic does not choose the right ones.
```

Fixes https://github.com/PeterJCLaw/srcomp-cli/issues/34

PeterJCLaw committed:
Warn when there are fewer teams than spaces in a schedule import

While this *is* supported, the logic around this somewhat guesswork
and it turns out to be non-obvious to users how to override the
default if that logic does not produce the desired result.

Fixes https://github.com/PeterJCLaw/srcomp-cli/issues/34

PeterJCLaw commented:
@samjmartin04 would be good to get your input here -- would the warning here have helped when you were looking at imports recently?

samjmartin04 commented:
Yeah I think that certainly would have helped. Perhaps mentioning that 3 robot matches should be given an id so they can be excluded rather than using something like null? Not a big problem either way though, that extra info with suggesting ignore ids makes it much better

PeterJCLaw commented:
For info: the ids here are strings, so `null` for example is fine "id" to use. Perhaps the docstring for the command needs to be clearer about that though :thinking:

samjmartin04 commented:
I suppose my thinking was more "you need to use exclude_ids with whatever you're using as your placeholder" as perhaps the assumption might be that null or similar is automatically excluded?

PeterJCLaw committed:
Extend docs on interpretation of ids in imported schedules

This aims to clarify the expectations here, some of which were
previously rather implicit.

PeterJCLaw commented:
@samjmartin04 does 67d4ac36691409b76e7b184e577590e95bfd40cd help?

samjmartin04 commented:
Yeah that looks good. I'd possibly add that ignoring certain ids is mostly used (or most useful) for making some 3 robot matches, but looks good to me

PeterJCLaw committed:
Note about coping with coprime match-entrant and overall-team counts
